airbyte_cdk/sources/declarative/extractors/response_to_file_extractor.py

```diff
@@ -89,21 +94,18 @@ def _save_to_file(self, response: requests.Response) -> Tuple[str, str]:
         """
         # set filepath for binary data from response
         decompressor = zlib.decompressobj(zlib.MAX_WBITS | 32)
-        needs_decompression = True  # we will assume at first that the response is compressed and change the flag if not
```

Comment:
As mentioned by the comment, we were relying on the response to know if we need to decompress. The logic of determining compression on the fly was implemented for sendgrid because of [this](https://github.com/airbytehq/airbyte/pull/35343). Is there a reason why we want to pass that at the initialization of the object? Have we validate that there won't be impact of sendgrid?

Reply:
my bad, i'll revert this

Reply:
reverted